presto-orc/src/main/java/io/prestosql/orc/metadata/Footer.java

```diff
@@ -45,7 +46,7 @@ public Footer(
             Map<String, Slice> userMetadata)
     {
         this.numberOfRows = numberOfRows;
-        this.rowsInRowGroup = rowsInRowGroup;
+        this.rowsInRowGroup = rowsInRowGroup == 0 ? OptionalInt.empty() : OptionalInt.of(rowsInRowGroup);
```

Comment:
Change the argument type instead. That will make it clear in the metadata reader that there is something special about zero.  Then lets verify the value in the optional is at least 1. Something like:
```
rowsInRowGroup.ifPresent(value -> checkArgument(value > 0, "rowsInRowGroup must be at least 1"));
```

Reply:
done